crates/kornia-imgproc/benches/bench_gradient.rs

```diff
@@ -0,0 +1,136 @@
+use criterion::{black_box, criterion_group, criterion_main, BenchmarkId, Criterion};
+
+use kornia_image::{Image, ImageError};
+use kornia_imgproc::filter::{
+    kernels, spatial_gradient_float, spatial_gradient_float_by_separable_filter, spatial_gradient_float_parallel
+};
+use rayon::{iter::{IndexedParallelIterator, ParallelIterator}, slice::ParallelSliceMut};
+
+/// Compute the first order image derivative in both x and y using a Sobel operator.
+///
+/// # Arguments
+///
+/// * `src` - The source image with shape (H, W).
+/// * `dst` - The destination image with shape (H, W, 2).
+pub fn spatial_gradient_float_rayon_parallel<const C: usize>(
+    src: &Image<f32, C>,
+    dx: &mut Image<f32, C>,
+    dy: &mut Image<f32, C>,
+) -> Result<(), ImageError> {
+    if src.size() != dx.size() {
+        return Err(ImageError::InvalidImageSize(
+            src.cols(),
+            src.rows(),
+            dx.cols(),
+            dx.rows(),
+        ));
+    }
+
+    if src.size() != dy.size() {
+        return Err(ImageError::InvalidImageSize(
+            src.cols(),
+            src.rows(),
+            dy.cols(),
+            dy.rows(),
+        ));
+    }
+
+    let (sobel_x, sobel_y) = kernels::normalized_sobel_kernel3();
+    let cols = src.cols();
+
+    let src_data = src.as_slice();
+
+    dx.as_slice_mut()
+        .par_chunks_mut(cols * C)
+        .zip(dy.as_slice_mut().par_chunks_mut(cols * C))
+        .enumerate()
+        .for_each(|(r, (dx_row, dy_row))| {
+            dx_row
+                .par_chunks_mut(C)
```

Comment:
can you try also without parallelization here ? most of the methods the trade off that works the best is only to do in parallel the rows iterator